### PR TITLE
feat: create topbar for edit submission form

### LINKED ISSUE #459: Create Navigation Bar Status template UI for the Moderation Panel
# Problem
Currently there is no way to save, reject, or approve the submissions. We need a component that just logs for now with the UI. Functionality can be added later

# Success
Looks the same as this photo

Image

### Requirements

- [ ] Has the same UI as the photo uploaded from Figma
- [ ] Logs when each button is clicked to the console, so those onClick events can have functionality later
- [ ] Contains a field has an ID and on click copies the ID to the clipboard. Only the ID value not the text `ID: `

# Implementation notes

### PR DESCRIPTION
**Pull request by NabbeunNabi**
Resolves #459 

## 🔧 What changed
Before there was no top navbar. Now there is partial functionality with styling matching that of the UI

## 🧪 Testing instructions
Go to the branch and go to the edit Submission part of the mod panel. When you see the topbar you can click the copy to clipboard. It should show you a check mark for 2 seconds to show that you successfully copied the id. There are comments about how to implement the autosave later. The Reject and Approve work with on click, but we need to create the call to the backend still within the store to put into those functions. But currently it logs. `You are rejected` or `You are accepted`

## 📸 Screenshots

-   ### Before
Nothing was there. Just text saying `TopBar Edit Submission`
Similar to this
image
 

-   ### After
Before clicking to copy
image

After clicking copy for 2 seconds
image

### CODE REVIEW

File: components/ModTopbar.vue [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     <!-- styling will change from px to rm -->
-    <div class="h-[76px] bg-emerald-100">
+    <div class="h-[76px] w-full">
         <div v-if="store.activeScreen === ModerationScreen.dashboard ">
             <p class="text-xl font-bold">TOPBAR MODERATION DASHBOARD</p>
         </div>
-        <div v-else-if="store.activeScreen === ModerationScreen.editSubmission">
-            <p class="text-xl font-bold">TOPBAR REVIEW SUBMISSION</p> 
+        <div class="h-[76px] w-full" v-else-if="store.activeScreen === ModerationScreen.editSubmission">
+           <ModEditSubmissionTopbar id="MattIsAGreatHumanBeing"/>
```
Comment (theyokohamalife) [inline components/ModTopbar.vue:8]: I agree that Matt is a great human being! Let's change this id to one that we can test later using `data-testid=""`

File: components/ModEditSubmissionTopbar.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                class="flex flex-row w-90 bg-neutral p-2 m-2 border-2 border-slate-400 rounded hover">
+                ID: {{ id }}
+                <SuccessCheckMark v-if="copySuccess" role="img" title="clipboard copy" class="ml-2 w-6" />
+                <CopyContent v-else role="img" title="successful copy" class="ml-2 w-6" />
+            </button>
+        </div>
+        <div class="flex flex-row justify p-2 font-bold ">
+            <button
+                class="flex justify-center items-center rounded-full bg-white border-slate-400 border-2 w-28 text-sm mr-2 ">Save
+                & Exit</button>
```
Comment (theyokohamalife) [inline components/ModEditSubmissionTopbar.vue:14]: I would add this string to the locale files. The "&" looks like it could potentially cause problems. Git doesn't seem too happy with it. 

Screenshot 2024-05-28 at 12 45 09 AM

File: components/ModTopbar.vue (hunk 1 above)
Comment (NabbeunNabi) [reply components/ModTopbar.vue:8]: That id there is a prop at the moment that we are passing but we will add a data test id as well!

File: components/ModEditSubmissionTopbar.vue (hunk 2 above)
Comment (NabbeunNabi) [reply components/ModEditSubmissionTopbar.vue]: Moved all the text except ID to the text.json file. No Japanese but has the translation string for now

File: components/ModEditSubmissionTopbar.vue [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const props = defineProps({
+    submissionId: {
+        type: String,
+        required: true,
+    }
+}
+)
+
+let copySuccess: Ref<boolean> = ref(false)
```
Comment (theyokohamalife) [inline components/ModEditSubmissionTopbar.vue:38]: I think we could make this variable name more specific as well. We should know from the name what it is that we're copying. Perhaps something like `copiedSubmissionId`?

File: components/ModTopbar.vue [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
-    <div class="h-[76px] bg-emerald-100">
-        <div v-if="store.activeScreen === ModerationScreen.dashboard ">
+    <div class="h-[76px] w-full">
+        <div v-if="store.activeScreen === ModerationScreen.dashboard">
             <p class="text-xl font-bold">TOPBAR MODERATION DASHBOARD</p>
         </div>
-        <div v-else-if="store.activeScreen === ModerationScreen.editSubmission">
-            <p class="text-xl font-bold">TOPBAR REVIEW SUBMISSION</p> 
+        <div class="h-[76px] w-full" v-else-if="store.activeScreen === ModerationScreen.editSubmission">
+            <ModEditSubmissionTopbar data-test-id="mod-edit-submission-topbar" submissionId="MattIsAGreatHumanBeing" />
```
Comment (theyokohamalife) [inline components/ModTopbar.vue:8]: I think we can initialize `submissionId` as a variable with an empty string below.

File: components/ModEditSubmissionTopbar.vue [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    navigator.clipboard.writeText(id).then(() => {
+        copySuccess.value = true;
+        setTimeout(() => {
+            copySuccess.value = false;
+        }, 2000);
+    }).catch(err => {
+        console.error('Failed to copy ID: ', err);
+    });
+}
```
Comment (theyokohamalife) [inline components/ModEditSubmissionTopbar.vue:51]: I think we should add a Cypress test for this behavior. You could write a test that clicks on the button and asserts that the expected checkmark is shown.

File: components/ModEditSubmissionTopbar.vue [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    submissionId: {
+        type: String,
+        required: true,
+    }
+}
+)
+
+let copySuccess: Ref<boolean> = ref(false)
+
+const copyId = () => {
```
[the remark this answers is not in the document]
Comment (NabbeunNabi) [reply components/ModEditSubmissionTopbar.vue]: Updated the name of this and other variables

File: components/ModEditSubmissionTopbar.vue [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                $t('modEditSubmissionTopNav.reject') }}</button>
+            <button @click="acceptSubmission"
+                class="flex justify-center items-center rounded-full bg-primary w-28 text-white text-sm mr-2 ">{{
+                $t('modEditSubmissionTopNav.approve') }}</button>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import CopyContent from '~/assets/icons/content-copy.svg'
```
Comment (ermish) [inline components/ModEditSubmissionTopbar.vue:26]: let's prefix the svg component names with `SVG*` to make it easy to identify

File: components/ModEditSubmissionTopbar.vue [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+}
+)
+
+let copySubmissionIdSuccess: Ref<boolean> = ref(false)
+
+const copySubmissionId = () => {
+    const submissionIdFromProps = props.submissionId
+
+    navigator.clipboard.writeText(submissionIdFromProps).then(() => {
+        copySubmissionIdSuccess.value = true;
```
Comment (ermish) [inline components/ModEditSubmissionTopbar.vue:44]: nit: remove the semicolons

File: components/ModEditSubmissionTopbar.vue [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import CopyContent from '~/assets/icons/content-copy.svg'
+import SuccessCheckMark from '~/assets/icons/checkmark-square.svg'
+import { Ref, ref } from 'vue';
+
+const props = defineProps({
+    submissionId: {
+        type: String,
+        required: true,
+    }
+}
```
Comment (ermish) [inline components/ModEditSubmissionTopbar.vue:35]: can you run the linter? I'm surprised these are on separate lines and also not tabbed

File: components/ModEditSubmissionTopbar.vue [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+let copySubmissionIdSuccess: Ref<boolean> = ref(false)
+
+const copySubmissionId = () => {
+    const submissionIdFromProps = props.submissionId
+
+    navigator.clipboard.writeText(submissionIdFromProps).then(() => {
+        copySubmissionIdSuccess.value = true;
+        setTimeout(() => {
+            copySubmissionIdSuccess.value = false;
```
Comment (ermish) [inline components/ModEditSubmissionTopbar.vue:46]: I was thinking it failed if it hit this timeout at first glance. I think we should rename the variable something like `showCopySuccessMessage` or something like that

**Review by ermish — CHANGES_REQUESTED** [review]
coming along!

File: components/ModEditSubmissionTopbar.vue (hunk 9 above)
Comment (NabbeunNabi) [reply components/ModEditSubmissionTopbar.vue:35]: Just ran the linter. It is setting it up the same way as the props in the `SearchResultsListItem.vue`

File: components/ModEditSubmissionTopbar.vue (hunk 10 above)
Comment (NabbeunNabi) [reply components/ModEditSubmissionTopbar.vue:46]: changed it to `showCopySuccessIcon` since it shows the SVG, not a message

**Review by theyokohamalife — APPROVED** [review]
LGTM

File: components/ModSubmissionListContainer.vue [hunk 11]
```diff
@@ -1,49 +1,56 @@
...
-    return `grid-template-columns: ${defaultColumnWidth}% repeat(${numColumns - 1}, ${columnWidth}%);`;
+    const columnWidth = remainingWidth / (numOfColumns - 1);
+    return `grid-template-columns: ${defaultColumnWidth}% repeat(${numOfColumns - 1}, ${columnWidth}%);`;
 });
 
-const accessSubmissionForm = (id:String) => {
-    // logic on how to pass the data to the submission form that is being edited
-}
+const handleClickToSubmissionForm = (id: string) => {
+    console.log("WE CLICKED THE THING!!")
```
Comment (ermish) [inline components/ModSubmissionListContainer.vue:50]: should remove if the logic is working now

File: i18n/locales/ja.json [hunk 12]
```diff
@@ -14,6 +14,37 @@
...
+        "about": "About Find a Doc, Japan",
+        "home": "Home",
+        "contact": "Contact",
+        "submit": "Submit",
+        "privacy": "Privacy",
+        "terms": "Terms",
+        "github": "Github",
+        "netlify": "Powered by Netlify"
+    },
+    "privacyPage": {
```
Comment (ermish) [inline i18n/locales/ja.json:27]: these changes should realistically be a separate PR

Comment (ermish) [reply i18n/locales/ja.json:27]: for the sake of time, I won't block on it, but realistically should try to make more atomic PRs

File: pages/moderation.vue [hunk 13]
```diff
@@ -12,11 +12,11 @@
 
 <script setup lang="ts">
 import { useModerationScreenStore, ModerationScreen } from "~/stores/moderationScreenStore"
-import { useRoute } from 'vue-router'
+// import { useRoute } from 'vue-router'
```
Comment (ermish) [inline pages/moderation.vue:15]: I think this code can be removed?

**Review by ermish — APPROVED** [review]
a few cleanup things and it looks good!

File: pages/moderation.vue (hunk 13 above)
Comment (NabbeunNabi) [reply pages/moderation.vue:15]: If you remove this code make sure all the other cypress tests are passing first. This is what was keeping the layout from shifting due to the env variable being implemented in development. Since the page layout is different for every other route once the env variable is introduced it can mess up the layout of every other page. I do not know either how on production we will make it know that it is on the moderation page to show the correct layout.

Comment (NabbeunNabi) [reply pages/moderation.vue:15]: Assuming this code isn't being utilized in the place it was previously being implemented

Comment (ermish) [reply pages/moderation.vue:15]: oh I was just referring to the commented out portion

File: i18n/locales/ja.json (hunk 12 above)
Comment (theyokohamalife) [reply i18n/locales/ja.json:27]: @ermish Agreed! These were only included, because we noticed they were missing while adding the new strings. I think we might want to implement an i18n plugin like we had in 1.0 to make the locale files easier to maintain.

File: pages/moderation.vue (hunk 13 above)
Comment (theyokohamalife) [reply pages/moderation.vue:15]: Actually, this file shouldn't change. Will undo them.
